pkg/gameservers/health.go

```diff
@@ -198,8 +198,28 @@ func (hc *HealthController) syncGameServer(ctx context.Context, key string) erro
 		return nil
 	}
 
-	if skip, err := hc.skipUnhealthy(gs); err != nil || skip {
-		return err
+	// retrieve the pod for the gameserver
+	pod, err := hc.podLister.Pods(gs.ObjectMeta.Namespace).Get(gs.ObjectMeta.Name)
+	if err != nil {
+		if k8serrors.IsNotFound(err) {
+			hc.baseLogger.WithField("gs", gs.ObjectMeta.Name).Debug("Could not find Pod")
+		} else {
+			// if it's something else, go back into the queue
+			return errors.Wrapf(err, "error retrieving Pod %s for GameServer to check status", gs.ObjectMeta.Name)
+		}
+	}
+
+	// If the pod doesn't exist, the GameServer is definitely not healthy
```

Comment:
This comment is confusing, because it's describing the "else" case for this conditional (not what is happening in the next block). What about:

"Make sure that the pod is in a healthy state"

Reply:
I agree it is confusing. How about renaming the `skipUnhealthy` into `skipUnhealthyGameContainer`.
This is to avoid the following (accentuating the confusion with comments) :
```go
	// Make sure that the pod is in a healthy state
	if pod != nil {
		// skip if unhealthy
		if skip, err := hc.skipUnhealthy(gs, pod); err != nil || skip {
			return err
		}

		// skip if healthy
		if !hc.isUnhealthy(pod) {
			hc.baseLogger.WithField("gs", gs.ObjectMeta.Name).WithField("podStatus", pod.Status).Debug("GameServer is not unhealthy anymore")
			return nil
		}
	}

```

I propose
```go
	// Make sure that the pod has to be marked unhealthy
	if pod != nil {
		if skip, err := hc.skipUnhealthyGameContainer(gs, pod); err != nil || skip {
			return err
		}

		// If the pod is not unhealthy anymore, go back in the queue
		if !hc.isUnhealthy(pod) {
			hc.baseLogger.WithField("gs", gs.ObjectMeta.Name).WithField("podStatus", pod.Status).Debug("GameServer is not unhealthy anymore")
			return nil
		}
	}
```

Reply:
lgtm!